app/resources/api/v2/specific_tube_rack_creation_resource.rb

```diff
@@ -0,0 +1,120 @@
+    #
+    # @note This resource cannot be modified after creation: its endpoint will not accept `PATCH` requests.
+    # @note Access this resource via the `/api/v2/specific_tube_rack_creations/` endpoint.
+    #
+    # Provides a JSON:API representation of {SpecificTubeRackCreation}.
+    #
+    # For more information about JSON:API see the [JSON:API Specifications](https://jsonapi.org/format/)
+    # or look at the [JSONAPI::Resources](http://jsonapi-resources.com/) package for Sequencescape's implementation
+    # of the JSON:API standard.
+    class SpecificTubeRackCreationResource < BaseResource
+      ###
+      # Attributes
+      ###
+
+      # @!attribute [w] parent_uuids
+      #   This is declared for convenience where the parent is not available to set as a relationship.
+      #   Setting this attribute alongside the `parents` relationship will prefer the relationship value.
+      #   @deprecated Use the `parents` relationship instead.
+      #   @param value [Array<String>] The UUIDs of labware that will be the parents for all tube racks
+      #   and tubes created.
+      #   @return [Void]
+      #   @see #parents
+      attribute :parent_uuids
+
+      def parent_uuids=(value)
+        @model.parents = value.map { |uuid| Labware.with_uuid(uuid).first }
+      end
+
+      # @!attribute [w] tube_rack_attributes
+      #   @param value [Array<Hash>] Hashes defining the attributes to apply to each tube rack and
+      #     the tubes within that are being created.
+      #     This is used to set custom attributes on the tube racks, such as name. As well as to create
+      #     the tubes within the tube rack and link them together.
+      #   @example [
+      #   {
+      #     :tube_rack_name=>"Tube Rack",
+      #     :tube_rack_barcode=>"TR00000001",
+      #     :tube_rack_purpose_uuid=>"0ab4c9cc-4dad-11ef-8ca3-82c61098d1a1",
+      #     :tube_rack_metadata_key=>"tube_rack_barcode",
+      #     :racked_tubes=>[
+      #       {
+      #         :tube_barcode=>"SQ45303801",
+      #         :tube_name=>"SEQ:NT749R:A1",
+      #         :tube_purpose_uuid=>"0ab4c9cc-4dad-11ef-8ca3-82c61098d1a1",
+      #         :tube_position=>"A1",
+      #         :parent_uuids=>["bd49e7f8-80a1-11ef-bab6-82c61098d1a0"]
+      #       },
+      #       etc... more tubes
+      #     ]
+      #   },
+      #   etc... more tube racks
+      #
+      #   @return [Void]
+      attribute :tube_rack_attributes
+
+      def tube_rack_attributes=(value)
+        return if value.nil?
+
+        # Convert ActionController::Parameters into hashes.
+        @model.tube_rack_attributes = value.map(&:to_unsafe_h)
+      end
+
+      # @!attribute [w] user_uuid
+      #   This is declared for convenience where the user is not available to set as a relationship.
+      #   Setting this attribute alongside the `user` relationship will prefer the relationship value.
+      #   @deprecated Use the `user` relationship instead.
```

Comment:
Adding something that is deprecated already? Am I misunderstanding?

Reply:
This is api v1 vs v2 legacy stuff, it will go with api v1 eventually. Copied from the other labwares. Ask Stuart for more info.